File: src/creational/factory/typescript/random-vehicle-algorithm.ts
import { BicycleFactory } from './classes/factories/bicycle-factory';
import { CarFactory } from './classes/factories/car-factory';
import { Vehicle } from './classes/interfaces/vehicle-protocol';

const carFactory = new CarFactory();
const bicycleFactory = new BicycleFactory();
const vehicles = [
  carFactory.getVehicle('Celta'),
  carFactory.getVehicle('Corsa'),
  bicycleFactory.getVehicle('Bicycle'),
];

const randomInt = () => Math.floor(Math.random() * vehicles.length);

export const randomCarAlgorithm = (): Vehicle => {
  return vehicles[randomInt()];
};

File: src/creational/factory/python/random_vehicle_algorithm.py
from random import randint

from classes.factories.car_factory import CarFactory
from classes.interfaces.vehicle_protocol import Vehicle

CAR_FACTORY = CarFactory()
BICYCLE_FACTORY = CarFactory()

VEHICLES = [
    CAR_FACTORY.get_vehicle('Celta'),
    CAR_FACTORY.get_vehicle('Corsa'),
    BICYCLE_FACTORY.get_vehicle('Bicycle'),
]


def randomCarAlgorithm() -> Vehicle:
    return VEHICLES[randint(0, len(VEHICLES) - 1)]
